plugin/pkg/scheduler/factory/factory.go

```diff
@@ -531,12 +531,21 @@ func (factory *ConfigFactory) makeDefaultErrorFunc(backoff *podBackoff, podQueue
 			}
 			// Get the pod again; it may have changed/been scheduled already.
 			pod = &api.Pod{}
-			err := factory.Client.Get().Namespace(podID.Namespace).Resource("pods").Name(podID.Name).Do().Into(pod)
-			if err != nil {
-				if !errors.IsNotFound(err) {
-					glog.Errorf("Error getting pod %v for retry: %v; abandoning", podID, err)
+			backoff := time.Duration(100*time.Millisecond)
+			for {
+				err := factory.Client.Get().Namespace(podID.Namespace).Resource("pods").Name(podID.Name).Do().Into(pod)
+				if err == nil {
+					break
+				}
+				if errors.IsNotFound(err) {
+					glog.Warning("A pod %v no longer exists", podID)
+					return
+				}
+				glog.Errorf("Error getting pod %v for retry: %v; retrying...")
+				backoff = backoff * 2
+				if backoff > time.Minute {
```

Comment:
We don't want super long backoff - we want to check it at least once per minute. This is pretty standard trick.